src/api/Type.js

```diff
@@ -0,0 +1,43 @@
+define(function () {
+    /**
+     * @typedef TypeDefinition
+     * @property {Metadata} metadata displayable metadata about this type
+     * @property {function (object)} [initialize] a function which initializes
+     *           the model for new domain objects of this type
+     * @property {boolean} [creatable] true if users should be allowed to
+     *           create this type (default: false)
+     */
+
+    /**
+     *
+     * @param {TypeDefinition} definition
+     * @constructor
+     */
+    function Type(definition) {
+        this.definition = definition;
+    }
+
+    /**
+     * Get a definition for this type that can be registered using the
+     * legacy bundle format.
+     * @private
+     */
+    Type.prototype.toLegacyDefinition = function () {
+        var def = {};
+        def.name = this.definition.metadata.label;
+        def.glyph = this.definition.metadata.glyph;
+        def.description = this.definition.metadata.description;
+
+        if (this.definition.initialize) {
```

Comment:
To verify that I understand correctly, you mean that `initialize` should be removed from TypeDefinition, and replaced with an `initialize` event fired whenever a new domain object is created. During that event, you can provide initial state for the model (or do whatever else you might want to do when a new object of that type is being created.) Is that right?

I like that approach, and it also extends well to other life cycle events if we need them (e.g. "whenever a Sine Wave Generator is deleted, bark like a dog")